### DISCUSSION TITLE
Any example of running a gradle task with the Launcher API?

### DISCUSSION THREAD

**Discussion by ghost** [junit-team/junit-framework#3719] [Q&A] [answered]
I would like to understand how I can run a Gradle task that will only run tests using the Launcher API.

Should this be a simple main method call? Or is there Gradle integration with the JUnit Platform?

**Comment by sbrannen** [comment, marked as answer]
> I would like to understand how I can run a Gradle task that will only run tests using the Launcher API.

Gradle actually uses the `Launcher` API itself to run tests on the JUnit Platform.

> Or is there Gradle integration with the JUnit Platform?

Yes, Gradle has built-in support for the JUnit Platform in the `test` task.

https://docs.gradle.org/current/dsl/org.gradle.api.tasks.testing.Test.html

> Should this be a simple main method call?

If you need something different than what Gradle's `test` task offers, yes, you could create your own `main()` method or custom Gradle task to interact with the `Launcher` API as you wish.
